### ISSUE TITLE
passenger-install-apache2-module fails to build on FreeBSD with versions 5.0.9 through 5.0.16

### ISSUE THREAD

**Issue by afair** [phusion/passenger#1586]
Complete Build Log and system versions available at: https://gist.github.com/afair/22e6664b5cb5d91a5da7

FreeBSD 8.4
Apache version: 2.2.29
Ruby 2.2.3 via chruby
Passenger 5.0.9 through 5.0.16 (5.0.8 is last working version)

Failed build command:

c++ -shared buildout/apache2/Configuration.o buildout/apache2/Bucket.o buildout/apache2/Hooks.o buildout/apache2/mod_passenger.o -fPIC -o buildout/apache2/mod_passenger.so   buildout/apache2/module_libpassenger_common/Logging.o buildout/apache2/module_libpassenger_common/Exceptions.o buildout/apache2/module_libpassenger_common/Utils/SystemTime.o buildout/apache2/module_libpassenger_common/Utils/StrIntUtils.o buildout/apache2/module_libpassenger_common/Utils/StrIntUtilsNoStrictAliasing.o buildout/apache2/module_libpassenger_common/Utils/IOUtils.o buildout/apache2/module_libpassenger_common/Utils.o buildout/apache2/module_libpassenger_common/ApplicationPool2/AppTypes.o buildout/apache2/module_libpassenger_common/Utils/MD5.o buildout/apache2/module_libpassenger_common/Utils/modp_b64.o buildout/apache2/module_libboost_oxt.a -L/usr/lib -Wl,-rpath,/usr/lib /usr/local/lib -fstack-protector -fPIC -I/usr/local/include/apr-1 -I/usr/local/include/apr-1 -I/usr/include -I/usr/local/include -I/usr/local/include/db5 -pipe -I/usr/include -fstack-protector -fno-strict-aliasing -I/usr/local/include/apache22 -L/usr/local/lib -lapr-1 -L/usr/local/lib -laprutil-1 -lpthread -lrt

    /usr/local/lib: file not recognized: File format not recognized
    rake aborted!

Looks like a malformed parameter. Should /usr/local/lib have a -L option before it so it will not be treated as a file to link? I was unable to trace the origin of this in the build system.

    -Wl,-rpath,/usr/lib /usr/local/lib -fstack-protector
                       ^

Let me know if you need more information, or need me to test something else.

... and Thank You for Passenger!

**Comment by FooBarWidget** [comment]
That indeed doesn't look correct. But I have the feeling that this is caused by your Apache installation. Can you tell me what these commands say?

    apr-1-config --cflags --link-ld
    apxs -q CFLAGS
    apxs -q LDFLAGS

**Comment by afair** [comment]
allen@dev2 % apr-1-config --cflags --link-ld
      -L/usr/local/lib -lapr-1
    allen@dev2 % apxs -q CFLAGS
    -O2 -pipe -I/usr/include -fstack-protector -fno-strict-aliasing
    allen@dev2 % apxs -q LDFLAGS
    -L/usr/lib -Wl,-rpath,/usr/lib /usr/local/lib -fstack-protector

Ahh, the LDFLAGS are giving the bad values. I know it is an older version of FreeBSD, Apache and gcc (4.2.1).

**Comment by FooBarWidget** [comment]
The latest version of FreeBSD we've tested on is 9. Versions before 8 are very old and too difficult for us to support. But at least for this issue we now know the culprit. :)

**Comment by shaiku** [comment]
What is the resolution?

**Comment by RealityDaemon** [comment]
I'm rocking with 10.2-RELEASE. Problem was similar for me I ran htop, and noticed that my compilation would run out of memory. I was installing the passenger module for apache at the time, and the installer was precompiling and stopping halfway due to an error. I found that after adding swap space, following the wonderful freebsd documentation below, re-compiling again was successful. 
I was of course able to add or extend the swap space without restarting my server. 

https://www.freebsd.org/doc/handbook/adding-swap-space.html

Cheers, RealityDaemon
